### PR TITLE
Event aggregates

### LINKED ISSUE #43: API to get event aggregates
create an API to get a filtered set of event aggregates - that'd be nice for analytics

we have to follow the principles set in https://github.com/adexnetwork/adex-protocol#privacy-of-publishersadvertisers

### PR DESCRIPTION
**Pull request by samparsky**
Fixes #43

### CODE REVIEW

File: routes/channel.js [hunk 1]
```diff
@@ -19,6 +19,9 @@ router.get('/:id/tree', channelLoad, getStatus.bind(null, true))
 router.get('/:id/validator-messages', getValidatorMessages)
 router.get('/:id/validator-messages/:uid/:type?', channelIfExists, channelLoad, getValidatorMessages)
 
+// event information
+router.get('/:id/events', authRequired, channelIfExists, channelLoad, getEvents)
```
Comment (Ivshti) [inline routes/channel.js]: we should rename this to event-aggregates, cause we're getting event aggregates

File: routes/channel.js [hunk 2]
```diff
@@ -48,6 +51,25 @@ function getStatus(withTree, req, res) {
...
+	const { uid } = req.session
+	const resp = { channel: req.channel }
+
+	const eventsCol = db.getMongo().collection('eventAggregates')
+	const key = `events.IMPRESSION.${uid}`
+
+	return eventsCol
+	.find({
+		[ key ]:  { "$exists": true }
+	})
```
Comment (Ivshti) [inline routes/channel.js]: you should project for the key here, in order to avoid leaking other publisher's data

File: routes/channel.js [hunk 3]
```diff
@@ -48,6 +51,25 @@ function getStatus(withTree, req, res) {
...
+	.limit(cfg.EVENTS_FIND_LIMIT)
+	.toArray()
+	.then(function(events) {
+		res.send({ ...resp, events })
```
Comment (Ivshti) [inline routes/channel.js]: also in the future this API should support aggregating the event aggregates by time periods; but let's leave this out of the PR

File: test/prep-db/mongo.js [hunk 4]
```diff
@@ -24,11 +25,18 @@ const dummyVals = {
 				{ id: 'awesomeFollower', url: 'http://localhost:8006' },
 			]
 		}
-	}
-
+	},
 }
 
-if (typeof(module) !== 'undefined') module.exports = dummyVals
+dummyVals['sessions'] = [
```
Comment (Ivshti) [inline test/prep-db/mongo.js]: this is definitely not needed, we dropped reading from the `sessions` collection in https://github.com/AdExNetwork/adex-validator-stack-js/commit/752feb140bb35a36704f04fda58344cf2523ad29

File: routes/channel.js (hunk 1 above)
Comment (samparsky) [reply routes/channel.js]: okay will change it

File: routes/channel.js (hunk 2 above)
Comment (samparsky) [reply routes/channel.js]: The other publisher data present would be the publisher balance and its exposed already on the blockchain

File: routes/channel.js (hunk 3 above)
Comment (samparsky) [reply routes/channel.js]: okay then

File: test/prep-db/mongo.js (hunk 4 above)
Comment (samparsky) [reply test/prep-db/mongo.js]: okay will make the change

File: routes/channel.js (hunk 2 above)
Comment (samparsky) [reply routes/channel.js]: okay done

**Pull request merged**